include/swaybar/config.h

```diff
@@ -65,6 +76,16 @@ uint32_t parse_position(const char *position);
 char *parse_font(const char *font);
 
 /**
+ * Parse display mode dock|hide|invisible.
+ */
+enum display_mode_types parse_display_mode(const char *display_mode);
+
+/**
+ * Parse hidden state show|hide.
+ */
+enum hidden_states parse_hidden_state(const char *hidden_state);
+
```

Comment:
>  take a hardline stance on macros. I will reject pull requests that use them

Regardless of context? I think that's wrong. Just because macros can be misused [doesn't mean](https://en.wikipedia.org/wiki/Ecological_fallacy) every particular use of them is misuse. cf. goto.  

> I've been pretty successful at making a codebase that's accessible to new contributors so far.

No argument here, but banning every use of macros had very little to do with it is my guess.

If you had 100 enums, would you prefer 100 boilerplate functions to a single macro?  Would that make for a cleaner codebase?

> I'd rather not establish a pattern that doesn't work everywhere
> It doesn't necessarily bring any significant benefits to the table in this case but it works for 
> more situations

Using the correct type of a variable also works in all cases and is certainly not an anti-pattern.

For the record, `sizeof` is a compile-time operator so you can use whatever index you like when sizing an array element. 0,1, 2**31-1, it doesn't matter (I overlooked that earlier).